### PR TITLE
SIE-47 Added horizontal-title component

### PR DESCRIPTION
**Pull request by quanganhhoang**
~~ Jira task [here](https://cs6510.atlassian.net/browse/SIE-47?atlOrigin=eyJpIjoiYWI0MjI4YTQxZDQ5NGY1NmJmNjllZDk3MzAzOGNmY2YiLCJwIjoiaiJ9) ~~

Screenshot: 
Screen Shot 2021-02-15 at 5 30 16 PM

### CODE REVIEW

**Comment by quanganhhoang** [comment]
Questions:
1.
> There should be a 50px gap between the title and paragraph; it could be decreased to 30px on tablets. Elements should be vertically stacked.

Is `50px` too big here? How would you prefer the padding/margin to be to display the gap?

2. Is `Work Sans` font installed yet?
3. Is there a template I should follow to add mobile responsive styling?

**Comment by alecherryy** [comment]
> Questions:
> 1.
> 
> > There should be a 50px gap between the title and paragraph; it could be decreased to 30px on tablets. Elements should be vertically stacked.
> 
> Is `50px` too big here? How would you prefer the padding/margin to be to display the gap?
> 
>     1. Is `Work Sans` font installed yet?
> 
>     2. Is there a template I should follow to add mobile responsive styling?

@quanganhhoang re:50px, you can set the gap at 50px for large tablets and above, after that it can shrink to 30px.
1. Work sans is being imported in the `_font.scss`, just make sure your component's stylesheet is importing the 'variables.scss` file. 
2. There isn't hard rule for responsive behavior; use your best judgment based on what looks good.

File: src/stories/components/HorizontalTitle/styles.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
+@import '../../../scss/variables.scss';
+
+// @file
+// HorizontalTitle
+
+.horizontal-title {
+    font-family: 'Work Sans';
+    padding-top: 0;
+    margin-top: 0;
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:9]: You don't need this. The `div` has no paddings nor margins by default

File: src/stories/components/HorizontalTitle/styles.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    &--medium {
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:11]: Love this! Never used it this way myself.

File: src/stories/components/HorizontalTitle/styles.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        .horizontal-title__title {
+            font-size: 20px;
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:13]: Font sizes are already declared in a global variables. Replace with `$font-size-lg`.

File: src/stories/components/HorizontalTitle/styles.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+// @file
+// HorizontalTitle
+
+.horizontal-title {
+    font-family: 'Work Sans';
+    padding-top: 0;
+    margin-top: 0;
+
+    &--medium {
+        .horizontal-title__title {
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:12]: `h1` font size should be the across across all component's instances. Remove this. What needs to change is the font size of the paragraph text.

File: src/stories/components/HorizontalTitle/styles.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    &--large {
+        .horizontal-title__title {
+            font-size: 24px;
+        }
+    }
+}
+
+.horizontal-title__title {
+    font-size: 16px;
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:24]: Remove this. Title does not change font size.

File: src/stories/components/HorizontalTitle/styles.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    padding-top: 0;
+    margin-top: 0;
+    padding-bottom: 10px;
+    margin-bottom: 30px;
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:28]: Use spacing global variable: `$spacing-1` and `$spacing-3`

File: src/stories/components/HorizontalTitle/styles.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.horizontal-title__eyebrow {
+    font-family: 'Work Sans';
+    font-size: 13px;
+    text-transform: uppercase;
+    letter-spacing: 1px;
+    padding-bottom: 0px;
+    margin: 0px;
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:37]: 1. Font family is being applied to `body` and is inherited by all of its children, so no need to declare it
2. Convert font-size to `rem` by doing `rem(13px)`
3. No need to declare margin and padding, the `div` has no padding nor margins by default

File: src/stories/components/HorizontalTitle/styles.scss [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.horizontal-title__content {
+    padding-top: 10px;
+    margin-top: 0;
+}
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/styles.scss:43]: No need to for this.

File: src/stories/components/HorizontalTitle/HorizontalTitle.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        <div
+            className={
+                ['horizontal-title', `${modifierClasses}`].join(' ').trim()
+            }
+        >
+            {eyebrow ?
+                <div className="horizontal-title__eyebrow">{eyebrow}</div> : ''
+            }
+            <h1 className="horizontal-title__title">{title}</h1>
+            <p className="horizontal-title__content">{content}</p>
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/HorizontalTitle.js:28]: Paragraphs shouldn't be nested inside of each other. Either convert this to a `div` or pass `{content}` as a string literal.

File: src/stories/components/HorizontalTitle/HorizontalTitle.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+ */
+export const HorizontalTitle = (
+    { modifierClasses, eyebrow, title, content }) => {
+    return (
+        <div
+            className={
+                ['horizontal-title', `${modifierClasses}`].join(' ').trim()
+            }
+        >
+            {eyebrow ?
```
Comment (alecherryy) [inline src/stories/components/HorizontalTitle/HorizontalTitle.js:24]: To get the horizontal layout, you'll probably want to put a wrapper `div` around the title and `eyebrow` and set the display of the component to grid or flex for large tablets and above. Example:

```
.horizontal-title {
        @include bp-tablet-lg {
            display: grid;
            grid-gap: $spacing-3;
        }

        @include bp-desktop {
            grid-gap: $spacing-5;
        }
    }
```

**Review by alecherryy — CHANGES_REQUESTED** [review]

**Comment by quanganhhoang** [comment]
Thanks for the detailed feedback! Much appreciate it! I addressed your comments @alecherryy

**Review by alecherryy — APPROVED** [review]

**Pull request merged**